src/actions/app.js

```diff
@@ -83,8 +64,12 @@ export function show404(url: string): Action {
   return { type: 'ROUTE_NOT_FOUND', url };
 }
 
-export function updateUrlState(urlState: UrlState): Action {
-  return { type: '@@urlenhancer/updateUrlState', urlState };
+export function updateUrlState(newUrlState: UrlState, state: State): Action {
+  return {
+    type: 'URL_STATE_UPDATED_FROM_BROWSER_NAVIGATION',
+    newUrlState,
+    state,
```

Comment:
Rather than this cryptic action, my biggest need here is to have access to the full previous state, so I know how to properly manage the update. I need this regardless of the greater changes and **can totally do this without rewriting the component**. However, I didn't know that until I rewrote it, and I prefer pulling this logic out of a React component, as it has nothing to do with our view layer. (This is where I want to solicit your opinion).